backend/spec/controllers/spree/admin/resource_controller_spec.rb

```diff
@@ -146,6 +153,21 @@ def model_class
     it 'destroys the resource' do
       expect { subject }.to change { Widget.count }.from(1).to(0)
     end
+
+    context 'failure' do
+      let(:widget) { Widget.create!(name: 'undestroyable') }
+      let(:params) { { id: widget.id } }
+
+      context 'js format' do
+        subject { delete :destroy, params: params, format: 'js' }
+
+        it 'responses with error message' do
+          subject
+          expect(response).to be_unprocessable
+          expect(response.body).to eq assigns(:widget).errors.full_messages.join(', ')
```

Comment:
Since you have the error hardcoded at [line 39](https://github.com/solidusio/solidus/pull/2920/files#diff-d39c34ac52242598fd564473683289eaR39) maybe it's better to check the expected result so that we are super sure we are testing the right thing, for example if both `assigns(:widget).errors.full_messages` and `response.body` are empty we could have this test passing even if it's not what we expect.

My suggestion is:

```ruby
expect(response.body).to eq 'You cant destroy undestroyable things!'
```

Reply:
The point is that I want to test that the answer contains exactly validation errors same as in other resource controller's actions
For example:
https://github.com/solidusio/solidus/blob/518765fd145b7072d80c141503cea2cce484793f/backend/spec/controllers/spree/admin/resource_controller_spec.rb#L97